test/cloudant.test.js

```diff
@@ -197,6 +200,93 @@ describe('cloudant connector', function() {
       });
   });
 
+  describe('model with array props gets updated properly', function() {
+    var prod1, prod2;
+    before('create Product', function(done) {
+      Product.create({
+        id: 1,
+        name: 'bread',
+        price: 100,
+        releases: [1, 2, 3],
+        type: ['plain', 'sesame', 'whole wheat'],
+        foo: [{id: 1, name: 'bread1'}, {id: 2, name: 'bread2'}],
+      }, function(err, product) {
+        if (err) return done(err);
+        prod1 = product;
+        Product.create({
+          id: 2,
+          name: 'bagel',
+          price: 100,
+          releases: [1, 2, 3],
+          type: ['plain', 'sesame', 'whole wheat'],
+          foo: [{id: 1, name: 'bagel1'}, {id: 2, name: 'bagel2'}],
+        }, function(err, product) {
+          if (err) return done(err);
+          prod2 = product;
+          done();
+        });
+      });
+    });
+
+    after(function(done) {
+      Product.destroyAll(function(err) {
+        if (err) return done(err);
+        done();
+      });
+    });
+
+    it('updates a single model with array props',
+      function(done) {
+        prod1.setAttribute('type', ['cinnamon raisin']);
+        prod1.setAttribute('releases', [4, 5, 6]);
+        prod1.setAttribute('foo', [{id: 3, name: 'bread3'}]);
+        Product.updateAll({id: 1}, prod1, {}, function(err, res) {
+          if (err) return done(err);
+          Product.findById('1', function(err, res) {
+            should.not.exist(err);
```

Comment:
- I think we might need more assertions to verify it's a full update(the original purpose of this issue)
- And could you also verify that if we update a specific field, the new data will override the old one? 
```js
prod1.setAttribute('foo', [{id: 1, name: 'newbread1'}, {id: 3, name: 'bread3'}]);
```
```js
var result = res._data;

result.releases.length.should.equal(3);
result. releases[0].should.equal(4);
result. releases[1].should.equal(5);
result. releases[2].should.equal(6);

result.type.length.should.equal(1); 
// this is what I am interested to know, 
// will cloduant reset other fields to null or totally delete them?
result.type[0].should.equal('cinnamon raisin');

result.foo.length.should.equal(2);
result.foo[the_index_of_id_3].name.should.equal('bread3');
result.foo[the_index_of_id_1].name.should.equal('newbread1');
```